src/transformers/integrations.py

```diff
@@ -926,74 +932,257 @@ def __del__(self):
-    """
-    A [`TrainerCallback`] that sends the logs to [Neptune](https://neptune.ai).
+    """TrainerCallback that sends the logs to [Neptune](https://neptune.ai).
+
+    Args:
+        api_token (`str`, optional):
+            Neptune API token obtained upon registration. You can leave this argument out if you have saved your token
+            to the `NEPTUNE_API_TOKEN` environment variable (strongly recommended). See full setup instructions in the
+            [docs](https://docs.neptune.ai/getting-started/installation).
+        project (`str`, optional):
+            Name of an existing Neptune project, in the form: "workspace-name/project-name". You can find and copy the
+            name from the project Settings -> Properties in Neptune. If None (default), the value of the
+            `NEPTUNE_PROJECT` environment variable will be used.
+        name (`str`, optional): Custom name for the run.
+        base_namespace (`str`, optional, defaults to "finetuning"): In the Neptune run, the root namespace
+            that will contain all of the logged metadata.
+        log_parameters (`bool`, optional, defaults to True):
+            If True, logs all Trainer arguments and model parameters provided by the Trainer.
+        log_checkpoints (`str`, optional, defaults to None):
+            If "same", uploads checkpoints whenever they are saved by the Trainer. If "last", uploads only the most
+            recently saved checkpoint. If "best", uploads the best checkpoint (among the ones saved by the Trainer). If
+            None, does not upload checkpoints.
+        run (`Run`, optional):
+            Pass a Neptune run object if you want to continue logging to an existing run. Read more about resuming runs
+            in the [docs](https://docs.neptune.ai/how-to-guides/neptune-api/resume-run).
+        **neptune_run_kwargs (optional):
+            Additional keyword arguments to be passed directly to the
+            [neptune.init_run()](https://docs.neptune.ai/api-reference/neptune#.init_run) function when a new run is
+            created.
     """
 
-    def __init__(self):
+    if TYPE_CHECKING and is_neptune_available():
+        from neptune.new.metadata_containers.run import Run
+
+    class MissingConfiguration(Exception):
+        def __init__(self):
+            super().__init__(
+                """
+            ------ Unsupported ---- We were not able to create new runs. You provided a custom Neptune run to
+            `NeptuneCallback` with the `run` argument. For the integration to work fully, provide your `api_token` and
+            `project` by saving them as environment variables or passing them to the callback.
+            """
+            )
+
+    INTEGRATION_VERSION_KEY = "source_code/integrations/transformers"
+    MODEL_PARAMETERS_KEY = "model_parameters"
+    TRIAL_NAME_KEY = "trial"
+    TRIAL_PARAMS_KEY = "trial_params"
+    TRAINER_PARAMETERS_KEY = "trainer_parameters"
+    FLAT_METRICS = {"train/epoch"}
```

Comment:
We use regular lowercased variable names for class variables, as is standard in Python.

Reply:
Sure. I wasn't certain as it was to indicate constants instead of some common variables. Done :wink: